### ISSUE TITLE
panic: reflect: Elem of invalid type postgres

### ISSUE THREAD

**Issue by Kiura** [go-pg/pg#215]
I pulled the current version of pg.v4 today, and my code is broken. I get the following error:

`panic: reflect: Elem of invalid type postgres`

Here is my code:
```
func (u *User) Read() error {
	_, err := db.QueryOne(u, `SELECT language, ... FROM users  WHERE id=?id`, u)
	return err
}
```

So when I go back to previous version, that I pulled few month, everything works as expected.

The full stack trace:
```
panic: reflect: Elem of invalid type

goroutine 21 [running]:
panic(0x47aac0, 0xc820158570)
	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
reflect.(*rtype).Elem(0x50e6c0, 0x0, 0x0)
	/usr/local/Cellar/go/1.6/libexec/src/reflect/type.go:599 +0x1c3
github.com/user/projectName/vendor/gopkg.in/pg.v4/types.ptrAppenderFunc(0xd94080, 0x50e6c0, 0x50dd60)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/types/append_value.go:76 +0x2b
github.com/user/projectName/vendor/gopkg.in/pg.v4/types.Appender(0xd94080, 0x50e6c0, 0xd94080)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/types/append_value.go:66 +0x141
github.com/user/projectName/vendor/gopkg.in/pg.v4/orm.newTable(0xd94080, 0x5c5c20, 0xc8202c16a8)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/orm/table.go:124 +0xeca
github.com/user/projectName/vendor/gopkg.in/pg.v4/orm.(*tables).Get(0xc820151980, 0xd94080, 0x5d3360, 0xd94080)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/orm/tables.go:31 +0xed
github.com/user/projectName/vendor/gopkg.in/pg.v4/orm.newStructTableModelValue(0x5d3360, 0xc8202c2000, 0x199, 0xc8202c1800, 0x0, 0x0)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/orm/model_table_struct.go:45 +0x383
github.com/user/projectName/vendor/gopkg.in/pg.v4/orm.newStructTableModel(0x606f00, 0xc8202b8100, 0xc8202b8100, 0x0, 0x0)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/orm/model_table_struct.go:28 +0x12e
github.com/user/projectName/vendor/gopkg.in/pg.v4/orm.NewModel(0xc8202a6260, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/orm/model.go:41 +0x911
github.com/user/projectName/vendor/gopkg.in/pg%2ev4.newSingleModel(0x5c5c20, 0xc8202c2000, 0xc8202c1a88, 0x0, 0x0)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/db.go:345 +0x11e
github.com/user/projectName/vendor/gopkg.in/pg%2ev4.(*DB).QueryOne(0xc8201efae0, 0x5c5c20, 0xc8202c2000, 0x47aac0, 0xc8202a6250, 0xc8202a6240, 0x1, 0x1, 0xc8202b40d0, 0x0, ...)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/vendor/gopkg.in/pg.v4/db.go:177 +0x47
github.com/user/projectName/models.(*User).Read(0xc8202c2000, 0x0, 0x0)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/models/user.go:113 +0x156
github.com/user/projectName/controllers.OnMessage(0x0, 0x0)
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/controllers/routes.go:140 +0x651
created by main.main
	/Users/user/Desktop/projects/0main/go/src/github.com/user/projectName/main.go:18 +0x57
exit status 2
```

**Comment by vmihailenco** [comment]
How `type postgres` is defined? How `User.ID` is defined?

**Comment by Kiura** [comment]
```
postgres := config.Conf.Postgres

db = pg.Connect(&pg.Options{
	User:     postgres.User,
	Password: postgres.Password,
	Database: postgres.Database,
	Addr:     postgres.Host + ":" + postgres.Port,
})
```

```
type User struct {
	ID           int64     `json:"id,omitempty,string"`
	Language     string    `json:"language,omitempty"`
	CreatedAt    time.Time `json:"-"`
	UpdatedAt    time.Time `json:"-"`
}

```

**Comment by vmihailenco** [comment]
Should be fixed in v4.1.8. Thanks for the report.

**Comment by Kiura** [comment]
Thx and great work )
